scripts/build-tools.sh

```diff
@@ -5,27 +5,71 @@
 # fail immediately on any errors
 set -e
 
-cd tools
-
-rm -rf build_tools
-
-mkdir build_tools
-cd build_tools
-cmake ..
-make -j$(nproc)
-for args in $@
-do
-	#build test topologies
-	if [[ "$args" == "-t" ]]; then
-		make tests -j$(nproc)
-	fi
-	#build fuzzer
-	if [[ "$args" == "-f" ]]; then
-		rm -rf build_fuzzer
-		mkdir build_fuzzer
-		cd build_fuzzer
-		cmake ../../fuzzer
-		make -j$(nproc)
-		cd ../
-	fi
-done
+SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)
```

Comment:
Duplicate SCRIPT_DIR definition, you have another one in main().

You can make all these `local` variables in `main()` if you prefer.

Reply:
`SCRIPT_DIR` is accessed in all the three functions, that's why made it global

Reply:
`local` in `main()` is almost the same as global. One difference is when you comment out `main()` and `source` the script in your current shell for interactive testing.
```diff

-main "$@"
+# main "$@"


```

Reply:
Ah, that's good to know. Very different than block scoping you'd see elsewhere. Using local now.